### PR TITLE
Fixes mpi thread specification in asteroid, generax and speciesrax

### PR DESCRIPTION
**Pull request by austinhpatton**
When the number of cpus specified to either of these three modules is the same as the number of vCPUs (threads), not cores, mpi will have issues. So, we now include the "--use-hwthread-cpus" flag which fixes this (threads are treated as cpus).

### CODE REVIEW

File: test/parameters.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (mertcelebi) [inline test/parameters.json:1]: this is no longer needed

**Review by mertcelebi — APPROVED** [review]
Remove the parameters file and merge away

File: modules/local/asteroid.nf [hunk 2]
```diff
@@ -29,7 +29,11 @@ process ASTEROID {
     def args = task.ext.args ?: ''
     """
     # Run asteroid using the multithreaded mpi version
-    mpiexec -np ${task.cpus} --allow-run-as-root asteroid -i $treefile -m $asteroid_map -p asteroid
+    mpiexec -np ${task.cpus} --allow-run-as-root --use-hwthread-cpus \
```
Comment (mertcelebi) [inline modules/local/asteroid.nf:32]: Do you need both `-np ${task.cpus}` and `--use-hwthread-cpus`?

File: test/parameters.json (hunk 1 above)
Comment (austinhpatton) [reply test/parameters.json:1]: Ah, yeah, thought I got rid of that....

File: modules/local/asteroid.nf (hunk 2 above)
Comment (austinhpatton) [reply modules/local/asteroid.nf:32]: i believe so: -np specifies the number of simultaneous processes to run, and --use-hwthread-cpus specifies that threads are treated as cpus. When calling mpiexec (or mpirun) you must specify the number of processes being run.

**Pull request merged**